### PR TITLE
Identity done

### PR DESCRIPTION
**Pull request by BohdanLytvynov**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Comment by BohdanLytvynov** [comment]
What have been done.
- Package Microsoft.AspNetCore.Identity.EntityFrameworkCore installed. (Due to the deprecation of Microsoft.AspNetCore.Identity)
- Old user class and it's repository and relating methods were removed from the project. Cause in Microsoft.AspNetCore.Identity.EntityFrameworkCore we have already build object to maintain Users.
- NoMoreUser migration was createed to drop user table.
- New Migration that has new Entityes was created (InitIdentity)
- Seeding database mechanism was updated.
- Service for using UserManager was configured.
- Database migration and seeding were tested localy. They are working good.

**Review by metrondir — APPROVED** [review]
lgtm

File: Streetcode/Streetcode.BLL/Mapping/Users/UserProfile.cs
Comment (vladtymo) [inline Streetcode/Streetcode.BLL/Mapping/Users/UserProfile.cs:1]: Add new user mapper configs

File: Streetcode/Streetcode.DAL/Entities/Users/User.cs [hunk 1]
```diff
@@ -1,23 +1,12 @@
...
 using Streetcode.DAL.Enums;
 
 namespace Streetcode.DAL.Entities.Users
 {
-    public class User
+    public class User : IdentityUser<Guid>
     {
-        public int Id { get; set; }
-
-        public string Name { get; set; } = string.Empty;
```
Comment (vladtymo) [inline Streetcode/Streetcode.DAL/Entities/Users/User.cs:11]: Add FirstName and LastName columns

File: Streetcode/Streetcode.WebApi/Program.cs [hunk 2]
```diff
@@ -32,7 +34,7 @@
 
 await app.ApplyMigrations();
 
-// await app.SeedDataAsync(); // uncomment for seeding data in local
+await app.SeedDataAsync(); // uncomment for seeding data in local
```
Comment (vladtymo) [inline Streetcode/Streetcode.WebApi/Program.cs:37]: Remove all changes related to seeding data

Comment (BohdanLytvynov) [reply Streetcode/Streetcode.WebApi/Program.cs:37]: Done

Comment (Careless-boop) [inline Streetcode/Streetcode.WebApi/Program.cs:37]: Comment this maybe

**Review by Careless-boop — APPROVED** [review]

File: Streetcode/Streetcode.WebApi/Program.cs (hunk 2 above)
Comment (Careless-boop) [reply Streetcode/Streetcode.WebApi/Program.cs:37]: And resolve the merge conflicts with UserRegisterDTO.

File: Streetcode/Streetcode.BLL/Mapping/Users/UserProfile.cs
Comment (BohdanLytvynov) [reply Streetcode/Streetcode.BLL/Mapping/Users/UserProfile.cs:1]: Done.
I have changed UserDto models. Created new one RegisterUseerDTO and connect it with other UserDTOs using Inherritance. Also new UserDTO was created for mapping to the Entity User
User Profile was updated.

File: Streetcode/Streetcode.DAL/Entities/Users/User.cs (hunk 1 above)
Comment (BohdanLytvynov) [reply Streetcode/Streetcode.DAL/Entities/Users/User.cs:11]: Done. New Fields added. Migration done

File: Streetcode/Streetcode.WebApi/Program.cs (hunk 2 above)
Comment (BohdanLytvynov) [reply Streetcode/Streetcode.WebApi/Program.cs:37]: Done

**Pull request merged**